src/app/components/channel-list/channel-list.component.ts

```diff
@@ -196,16 +213,16 @@ export class ChannelListComponent implements OnInit, OnDestroy, AfterViewInit {
             return bBalance.minus(aBalance).toNumber();
         });
 
-        if (this.showAll) {
-            this.visibleChannels = filteredChannels;
-        } else {
-            this.visibleChannels = filteredChannels.slice(0, this.itemsPerRow);
-        }
+        this.visibleChannels = filteredChannels;
     }
 
     private calculateItemsPerRow() {
-        const listWidth = this.listElement.nativeElement.getBoundingClientRect()
+        const sectionWidth = this.channelsElement.nativeElement.getBoundingClientRect()
             .width;
-        this.itemsPerRow = Math.floor((listWidth - 213) / 229);
+        this.itemsPerRow = Math.floor(
```

Comment:
Isn't there any `flow` way to do this? Doing it by hand seems error prone, although it's ok here

Reply:
Yes would be nicer by CSS. But I think there is no way of positioning the items with the same width for all, with a fixed space between and using the full container width. With the `flex-wrap` that we had before, there was space left on the right side.